lib/Runtime/StatsExporter.cpp

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Runtime/StatsExporter.h"
+
+#include <vector>
+
+namespace glow {
+
+static std::vector<StatsExporter *> &getExporters() {
+  static std::vector<StatsExporter *> exporters;
+  return exporters;
+}
+
+void StatsExporter::registerStatsExporter(StatsExporter *exporter) {
+  getExporters().push_back(exporter);
+}
+
+void StatsExporter::addQuantileValue(llvm::StringRef key, double value) {
+  for (auto const &exporter : getExporters()) {
+    exporter->addQuantileValue(key, value);
+  }
+}
+
+void StatsExporter::setCounter(llvm::StringRef key, int64_t value) {
+  for (auto const &exporter : getExporters()) {
+    exporter->setCounter(key, value);
+  }
+}
+
+void StatsExporter::incrementCounter(llvm::StringRef key, int64_t value) {
+  for (auto const &exporter : getExporters()) {
+    exporter->incrementCounter(key, value);
+  }
+}
+
+StatsExporter Stats;
```

Comment:
Might cause some destruction order problems right?
What if some other things destructor calls `Stats.incrementCounter("foo" -1)` for example but this is already destroyed? Maybe make it a pointer?

Reply:
Good catch, I only thought about SIOF and not SDOF :-p